docs/sources/reference/components/discovery/discovery.process.md

```diff
@@ -29,6 +29,7 @@ The following arguments are supported:
 | Name               | Type                | Description                                                                              | Default | Required |
 |--------------------|---------------------|------------------------------------------------------------------------------------------|---------|----------|
 | `join`             | `list(map(string))` | Join external targets to discovered processes targets based on `__container_id__` label. |         | no       |
+| `cgroup_id_regex`  | `string`            | Regular expression that captures only one group that will used as cgroup ID of process.  |         | no       |
```

Comment:
It may be a dumb question, but is it not possible to use `discovery.relabel` instead of introducing `cgroup_id_regex`  ?

Reply:
Ahh I see what you mean but even to use relabel we need `process` component exporting cgroup path field in the targets. Because we need a cgroup ID which maps to a compute unit (pod, VM or batch job) of resource manager. which is usually in the cgroup path.

But I guess your suggestion opens a new direction. We can simply export cgroup path as one of the target fields and end users can use `relabel` to retrieve cgroup ID from the path. This will simplify error handling as well as there is no regex compiling involved and so, no need of updating and any mutexes. 

@mattdurham What do you think?

Reply:
@mattdurham Just checking in to see where are we with this PR.

Reply:
>  We can simply export cgroup path as one of the target fields and end users can use relabel to retrieve cgroup ID from the path.

This would be very helpful. I would like to set service_name based on the systemd service name (which could be determined by cgroup). Looking forward have this PR merged